compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -805,23 +797,32 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             }
             return false;
         };
+        let generate_new = |orig_ty, old_ref: &ty::Binder<'tcx, ty::TraitRef<'tcx>>| {
```

Comment:
Looks like `orig_ty` doesn't need to be passed either?
```
let orig_ty = old_ref.self_ty().skip_binder();
```
Then this whole logic of `generate_new` can be inlined into `try_borrowing`.